core/unit_test/TestTeam.hpp

```diff
@@ -1616,6 +1616,69 @@ struct TestTeamPolicyHandleByValue {
 
 }  // namespace
 
+namespace {
+template <typename ExecutionSpace>
+struct TestRepeatedTeamReduce {
+  using TeamPolicy          = Kokkos::TeamPolicy<ExecutionSpace>;
+  static constexpr int ncol = 1500;  // nothing special, just some work
+
+  KOKKOS_FUNCTION void operator()(
+      const typename TeamPolicy::member_type &team) const {
+    // non-divisible by power of two to make triggering problems easier
+    constexpr int nlev = 129;
+    constexpr auto pi  = Kokkos::numbers::pi;
+    double b           = 0.;
+    for (int ri = 0; ri < 10; ++ri) {
+      const auto g1 = [&](const int k, double &acc) {
+        acc += Kokkos::cos(pi * double(k) / nlev);
```

Comment:
I think that using trigonometric functions is unnecessary complexity in that test

Reply:
I would prefer to keep it since it was in the original reproducer unless we have a better idea.

Reply:
Are you not able to trigger the issue with a simple reduction that sums ones?

Reply:
I could imagine that the cos call stuff makes the threads get out of sync, because they funnel through limited special function units?

Reply:
I can't reproduce the failure with something like
```
    const auto g1 = [&](const int k, double &acc) {
        acc += 1. / nlev;
      };
      const auto g2 = [&](const int k, double &acc) {
        acc += .5 / nlev;
      };
      double a1, a2;
      Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, nlev), g1, a1);
      Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, 2*nlev), g2, a2);
```

Reply:
Please add a comment to that effect in the code

Reply:
Thanks for checking!

Reply:
[`087d767` (#5540)](https://github.com/kokkos/kokkos/pull/5540/commits/087d767855d571dd42a8818c67b0fac2d4862158)